### PR TITLE
fix: adding emergency garbage collection at 90% for chromium-based browsers

### PR DESCRIPTION
**Pull request by mschile**
Edit: Reverted in https://github.com/cypress-io/cypress/pull/25546

- Closes n/a

### User facing changelog

n/a

### Additional details

In an effort to reduce OOM errors, we decided to perform a GC when the browser's renderer process reaches 90% of the available memory.

### How has the user experience changed?

n/a

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/s] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/server/lib/browsers/memory/index.ts [hunk 1]
```diff
@@ -221,12 +228,20 @@ export const calculateMemoryStats: () => Promise<void> = measure(async () => {
...
 
   // if we should collect garbage, set the flag to true so we can collect garbage on the next test
   collectGarbageOnNextTest = collectGarbageOnNextTest || shouldCollectGarbage
 
+  // if the renderer's memory is above the EMERGENCY_MEMORY_THRESHOLD_PERCENTAGE, we should perform an emergency garbage collection now
+  const shouldEmergencyCollectGarbage = rendererUsagePercentage >= EMERGENCY_MEMORY_THRESHOLD_PERCENTAGE && !SKIP_GC
+
+  if (shouldEmergencyCollectGarbage) {
+    debug('emergency garbage collection triggered')
+    await checkMemoryPressure(automation, shouldEmergencyCollectGarbage)
```
Comment (mjhenkes) [inline packages/server/lib/browsers/memory/index.ts:242]: Does this reset `shouldCollectGarbage`?

**Review by chrisbreiding — APPROVED** [review]

File: packages/server/lib/browsers/memory/index.ts (hunk 1 above)
Comment (mschile) [reply packages/server/lib/browsers/memory/index.ts:242]: No, I debated whether or not I should clear it, and decided not to. Thoughts?

Comment (mjhenkes) [reply packages/server/lib/browsers/memory/index.ts:242]: If we don't reset, we could be collecting garbage on the next test even if the threshold is below 50 percent right?
The next cycle of checking would enable collect garbage again if we didn't drop below 50% right?

Comment (mjhenkes) [reply packages/server/lib/browsers/memory/index.ts:242]: RIGHT? sorry i said right too many times in my last comment 😅

Comment (mschile) [reply packages/server/lib/browsers/memory/index.ts:242]: Right :) Though if it didn't drop below threshold and the next test started prior to the next interval we would be missing a GC. I decided to err on the side of more GCs.

Comment (mjhenkes) [reply packages/server/lib/browsers/memory/index.ts:242]: 🤷‍♂️, fair point.

**Review by mjhenkes — APPROVED** [review]

**Pull request merged**